File: customview/src/main/java/com/example/customview/adapter/MainAdapter.java
package com.example.customview.adapter;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.customview.R;
import com.example.customview.bean.TypeBean;

import java.util.List;
/**
 * 用于适配 Main 数据的适配器。
 */

public class MainAdapter extends BaseQuickAdapter<TypeBean, BaseViewHolder> {

    public MainAdapter(@Nullable List<TypeBean> data) {
        super(R.layout.adapter_item_main, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, TypeBean item) {
        helper.setText(R.id.title_tv, item.getTitle());
    }
}

File: customview/src/main/java/com/example/customview/adapter/MainAdapter.kt
package com.example.customview.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.example.customview.bean.TypeBean
import com.example.customview.databinding.AdapterItemMainBinding

/**
 * 用于适配 Main 数据的适配器。
 *
 * Item 的创建和数据绑定都直接使用 ViewBinding，避免 BaseViewHolder.setText()
 * 在运行时通过 findViewById 查找不到控件而产生空指针异常。
 */
open class MainAdapter(data: List<TypeBean>?) :
    BaseQuickAdapter<TypeBean, BaseViewHolder>(data) {

    override fun onCreateDefViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): BaseViewHolder {
        val binding = AdapterItemMainBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return BaseViewHolder(binding.root)
    }

    override fun convert(helper: BaseViewHolder, item: TypeBean) {
        val binding = AdapterItemMainBinding.bind(helper.itemView)
        binding.titleTv.text = item.title
    }
}
